### PR TITLE
Wizard message UI improvement

### PR DESCRIPTION
**Pull request by alanrenmsft**
Fix for issue #2662 

API Changes:
Added a new optional description property for the DialogMessage class, this will be displayed in a smaller font in the message box. this can be used for stack trace or long text complementary for the text property.

UI Changes:
Instead of showing the message in the footer, now the message will be displayed in the top of wizard with a max height to fit in 5 lines of text, a scrollbar will appear if it goes beyond 5 lines.

One line text:
image

long text - Collapsed (default) mode:
image

long text - Expanded mode:
image

Icon on the left side: indicating the severity (level property of DialogMessage class)

Details button: when the text is too long or if the description property is set, the button will become visible to expand/collapse the message box

Copy button: copy the text as well as the description to clipboard, a separator line will be added between them just like SSMS.

Close button: close the message box.

Message label: used to display the text property of the DialogMessage class.
    - If the text is too long to fit in one line, an ellipsis will be displayed at the end
    - If the text is too long to fit in one line or if the description property is set, the label will become clickable with pointer cursor, behavior is the same as Details button

The theme of the message box will be different based on the level property of DialogMessage class.

Warning:

image

Info:
image

### CODE REVIEW

File: src/sql/platform/dialog/media/wizardNavigation.css [hunk 1]
```diff
@@ -7,8 +7,7 @@
 	display: flex;
 	flex-direction: column;
 	width: 80px;
-	height: calc(100% + 25px);
-	margin-top: -25px;
```
Comment (MattIrv) [inline src/sql/platform/dialog/media/wizardNavigation.css]: Great to get rid of the `calc` heights and hacky margins here and in the other css files above. Thanks!

File: src/sql/base/browser/ui/modal/modal.ts [hunk 2]
```diff
@@ -82,6 +97,7 @@ export abstract class Modal extends Disposable implements IThemable {
 
 	private _modalDialog: Builder;
 	private _modalHeaderSection: Builder;
+	private _modalMessageSecion: Builder;
```
Comment (MattIrv) [inline src/sql/base/browser/ui/modal/modal.ts]: typo: 'section'

**Comment by alanrenmsft** [comment]
Thanks @MattIrv for the comments, I have addressed all of them and pushed the changes.

image

image

File: src/sql/base/browser/ui/modal/modal.ts (hunk 2 above)
Comment (alanrenmsft) [reply src/sql/base/browser/ui/modal/modal.ts]: fixed

**Pull request merged**